### PR TITLE
Await apiHandler to handle error in try/catch

### LINKED ISSUE #1598: ErrorHandlerMiddleware is ignore with 6.1.*
## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

When an error is thrown, custom ErrorHandlerMiddleware catches the error and returns a response with according errorCode (400, 403, 429, etc.). Works as expected with 6.0.*

## Current Behavior

When an error is thrown, custom ErrorHandlerMiddleware is ignore and app hangs up, nothing happens. Breaks with upgrade to 6.1.4 and 6.1.5, and routes regenerated.

## Context (Environment)

Version of the library: 6.1.*
Version of NodeJS: 20.*

- Confirm you were using yarn not npm: [ ]

## Breaking change?

It seems that is related to the new way of generated routes that handles requests.
I have passed many hours to debug, but does not find something really more helpfull. Sorry !

### PR DESCRIPTION
**Pull request by VincentClair**
### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/lukeautry/tsoa/tree/master/docs/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/lukeautry/tsoa/pulls) for the same update/change?
- [x] Have you written unit tests?
- [x] Have you written unit tests that cover the negative cases (i.e.: if bad data is submitted, does the library respond properly)?
- [x] This PR is associated with an existing issue?

**Closing issues**

closes #1598

**Test plan**

Checks that a controller method without `async` keyword throwing an error is caught.

The real problem is if method is not a promise (`async` keyword), and error throw immediately when building promise, server is no more responding, and we get a time out (actually 5sec in tests).

The problem could resolve in two ways : 
- put the `buildPromise` int the try/catch  https://github.com/lukeautry/tsoa/blob/efdb468a5cedcb9bef345f1eed776e8fc621b443/packages/runtime/src/routeGeneration/templates/express/expressTemplateService.ts#L41
- wait for `apiHandler` to be resolved on generated template https://github.com/lukeautry/tsoa/blob/efdb468a5cedcb9bef345f1eed776e8fc621b443/packages/cli/src/routeGeneration/templates/express.hbs#L106
I have chosen to apply both, as I think code will be more robust.

### CODE REVIEW

File: packages/cli/src/routeGeneration/templates/express.hbs [hunk 1]
```diff
@@ -103,7 +103,7 @@ export function RegisterRoutes(app: Router) {
                 const controller = new {{../name}}();
               {{/if}}
 
-              templateService.apiHandler({
```
Comment (WoH) [inline packages/cli/src/routeGeneration/templates/express.hbs:106]: I think we can always await, making the apiHandler return A | Promise<A>

Comment (VincentClair) [reply packages/cli/src/routeGeneration/templates/express.hbs:106]: agree, but function must be always async: we could replace condition `{{#if ../../iocModule}}async {{/if}}`line 82 by async directly.

File: packages/runtime/src/routeGeneration/templates/express/expressTemplateService.ts [hunk 2]
```diff
@@ -38,9 +38,9 @@ export class ExpressTemplateService extends TemplateService<ExpressApiHandlerPar
 
   async apiHandler(params: ExpressApiHandlerParameters) {
     const { methodName, controller, response, validatedArgs, successStatus, next } = params;
-    const promise = this.buildPromise(methodName, controller, validatedArgs);
 
     try {
+      const promise = this.buildPromise(methodName, controller, validatedArgs);
       const data = await Promise.resolve(promise);
```
Comment (VincentClair) [inline packages/runtime/src/routeGeneration/templates/express/expressTemplateService.ts:44]: is Promise.resolve mandatory : promise ca be waited directly, no? `const data = await promise;`

File: packages/cli/src/routeGeneration/templates/express.hbs (hunk 1 above)
Comment (WoH) [reply packages/cli/src/routeGeneration/templates/express.hbs:106]: Yeah, that was the ultimate goal, while still making ts not complain about it.

Comment (jackey8616) [reply packages/cli/src/routeGeneration/templates/express.hbs:106]: > I think we can always await @jackey8616

Yes, I think that's good.

Comment (VincentClair) [reply packages/cli/src/routeGeneration/templates/express.hbs:106]: done

**Comment by VincentClair** [comment]
Hello @WoH @jackey8616,
How can i help to validate this PR ? Is there something missing ?

**Review by WoH — APPROVED** [review]
No, lgtm 🚀

**Comment by denizdogan** [comment]
When can we expect a new release with this fix included? Much appreciated

**Pull request merged**